### ISSUE TITLE
ansible-lint is not supported on OpenShift with rhel-system-roles on RHEL 8

### ISSUE THREAD

**Issue by realtimetodie** [ansible/ansible-lint#3251]
##### Summary

When running ansible-lint on RedHat OpenShift inside containers using RHEL 8, the [rhel-system-roles](https://access.redhat.com/articles/3050101) playbook is not being recognized. Instead, many verbose and annoying warning messages are thrown by ansible-lint. Finally, the ansible-lint process exits with an unknown error.

It is unclear to us, how to install ansible-lint on RHEL 8 containers running on RedHat OpenShift properly as this is nowhere documented.

There is an `ansible-lint` rpm package that can be installed using dnf. However, the ansible-lint rpm package can only be installed on environments where the RedHat subscription-manager is enabled. The subscription manager is disabled on RedHat OpenShift containers by default.

Example (does not work on ~~RedHat OpenShift~~)

```bash
$ dnf install ansible-lint
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```bash
$uname -a
Linux ansible-lint-test-zt42k 4.18.0-305.76.1.el8_4.x86_64 #1 SMP Thu Jan 12 10:05:36 EST 2023 x86_64 x86_64 x86_64 GNU/Linux
```

```bash
$ ansible-lint --version
ansible-lint 6.14.4 using ansible 2.13.3
```

##### STEPS TO REPRODUCE

Dockerfile

```
FROM registry.access.redhat.com/ubi8:8.6-903

RUN dnf -y upgrade --security && \
    dnf -y --setopt=skip_missing_names_on_install=False install rhel-system-roles python39 python39-pip && \
    alternatives --set python /usr/bin/python3.9 && \
    pip3 install --upgrade pip && \
    pip3 install ansible-lint
```

Playbook collections/requirements.yml

```yaml
---
collections:
  - name: redhat.rhel_system_roles
```

Run ansible lint inside the container

```bash
$ ansible-lint -r .
```

```bash
WARNING  /usr/lib/python3.9/site-packages/ansible/_vendor/__init__.py:42: UserWarning: One or more Python packages bundled by this ansible-core distribution were already loaded (packaging). This may result in undefined behavior.
  warnings.warn('One or more Python packages bundled by this ansible-core distribution were already '

WARNING  Retrying execution failure 1 of: ansible-galaxy collection install -v -r collections/requirements.yml -p /test
ERROR    Using /etc/ansible/ansible.cfg as config file
Starting galaxy collection install process
Process install dependency map

Warning: [WARNING]: The specified collections path '/test' is not part of the configured Ansible collections
paths '/test'. The
installed collection won't be picked up in an Ansible run.
ERROR! Failed to resolve the requested dependencies map. Could not satisfy the following requirements:
* redhat.rhel_system_roles:* (direct request)

Got 1 exit code while running: ansible-galaxy collection install -v -r collections/requirements.yml -p /test/collections
```

##### Desired Behavior

ansible-lint should work with the rhel-system-roles package on OpenShift.

##### Actual Behavior

ansible-lint does not work when running on OpenShift and the rhel-system-roles package is installed.

##### Related

- https://access.redhat.com/articles/3050101

**Comment by ssbarnea** [comment]
For some reason I am unable to install this rpm on rhel 9 and I do not have the time to install another VM for testing:
```
[root@r9 ~]# dnf search rhel-system-roles
Updating Subscription Management repositories.
Last metadata expiration check: 0:08:17 ago on Thu 13 Apr 2023 11:10:24 AM BST.
========================== Name Exactly Matched: rhel-system-roles ==========================
rhel-system-roles.noarch : Set of interfaces for unified system management
[root@r9 ~]# dnf install rhel-system-roles
Updating Subscription Management repositories.
Red Hat Enterprise Linux 9 for ARM 64 - AppStream (RPMs)     314  B/s | 361  B     00:01
Errors during downloading metadata for repository 'rhel-9-for-aarch64-appstream-rpms':
  - Status code: 403 for https://cdn.redhat.com/content/dist/rhel9/9/aarch64/appstream/os/repodata/repomd.xml (IP: 2.22.12.251)
Error: Failed to download metadata for repo 'rhel-9-for-aarch64-appstream-rpms': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

As this rpm should be installable on any machine, please send me a link that I can use to install it directly by email or private chat.

Apparently my r9 test box have some subscription-manager issues,...

**Comment by realtimetodie** [comment]
@linux-system-roles @bachradsusi @jamacku @pcahyna @sergio-correia @tyll @wrabcak

Please, we need your help. Apparently, [rhel-system-roles](https://access.redhat.com/articles/3050101) does not work with ansible-lint on Red Hat OpenShift.

**Comment by pcahyna** [comment]
`ansible-lint --help` shows this option:
```
  --offline             Disable installation of requirements.yml
```
  -- try it.

**Comment by realtimetodie** [comment]
Thank you @ssbarnea 💪
